### PR TITLE
Azure SDK packages page

### LINKED ISSUE #19521: Azure .NET NuGet Package list needs to be hosted on Docs
- [x] Create a .NET page like [this](https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-library-package-index) in the Azure Developer docs repo - Done, [PR](https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/818)

- [ ] Create context object and contextual TOC pointing there from /dotnet/azure

### PR DESCRIPTION
**Pull request by CamSoper**
Implements an up-to-date package list from the Azure SDK team. Resolves #19521 .

### CODE REVIEW

File: docs/azure/packages.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: Package index for Azure SDK for .NET
+description: Listing of the package names, NuGet links, docs links, and source code links for all libraries in the Azure SDK for .NET.
+ms.service: multiple
+ms.custom: azure-sdk-dotnet
```
Comment (Youssef1313) [inline docs/azure/packages.md:5]: There is a build warning here.

> Only one of the following attributes can exist: 'ms.prod', 'ms.service'. Use ms.prod for on-premise products, or ms.service for cloud services.

Comment (Youssef1313) [reply docs/azure/packages.md:5]: `ms.prod` is defined in [docfx.json](https://github.com/dotnet/docs/blob/e20efe7eb096f27337081bfb9a2ef2304eac59f8/docfx.json#L168).

File: docs/azure/packages.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+author: camsoper
+ms.author: casoper
+ms.date: 07/29/2020
+---
+
+# Azure SDK for .NET package index
+
+[Libraries using Azure.Core](#libraries-using-azurecore)
+
+[All libraries](#all-libraries)
```
Comment (tdykstra) [inline docs/azure/packages.md:14]: These duplicate the right nav menu; could replace them with brief text describing what's in the doc and what it's for.

**Review by tdykstra — APPROVED** [review]
LGTM; one suggestion.

**Pull request merged**